src/components/EditorCardsDrawer/EditorCardsDrawer.tsx

```diff
@@ -17,21 +93,244 @@ export const EditorCardsDrawer = ({
   onClose,
   onProceed,
 }: EditorCardsDrawerProps): JSX.Element => {
+  const [previewState, setPreviewState] = useState<EditorCard[]>([])
+  const [initialEditorState, setInitialEditorState] = useState<EditorCard[]>([])
+  const methods = useForm<EditorCardsInfo>({
+    mode: "onChange",
+    resolver: yupResolver(editorCardsInfoSchema),
+    defaultValues: {
+      isDisplayImage: true,
+      cards: [],
+    },
+  })
+
+  const handleSubmit = (data: EditorCardsInfo) => {
+    editor.commands.setCardsContent(getEditorCardsContent(data))
+    onProceed()
+  }
+
+  const handleClose = () => {
+    editor.commands.setCardsContent(initialEditorState)
+    onClose()
+  }
+
+  const onDragEnd: OnDragEndResponder = ({ source, destination }) => {
```

Comment:
note: The `onDragEnd`, `onCreate` and `onDelete` methods have been updated in #1707 to take advantage of react-hook-form's useFieldArray method, which has better performance and native support.